erpnext/stock/doctype/purchase_receipt/purchase_receipt.py

```diff
@@ -180,6 +170,8 @@ def on_submit(self):
 
 		self.make_gl_entries()
 		self.update_package_tag_batch()
+		update_per_received_and_status_in_pp(self)
```

Comment:
@MyuddinKhatri You modified this from the status updater. Then make sure it should call on on_submit
and check the scenario when you cancel purchase receipt  it should make it material request status back requested

Reply:
removed from status_updater and added in this function and production plan set_status() is called on PR submit so it sets Material Requested accordingly as it was working previously

Reply:
@MyuddinKhatri check  for on_cancel scenario